### PR TITLE
CO's Deagle and Mateba Rebalance

### PR DESCRIPTION
**Pull request by nauticall**
## About The Pull Request

Rebalances the CO's weapons, the Desert Eagle and Mateba, to be able to play off each other's strengths instead of one being statistically better than the other in every aspect.

**Both** weapons have received a rebalance in ammo distribution. **All ammunition is now high-impact capable**, being able to stun and perform battlefield executions if the wielder is capable of doing so. 

Both weapons have high-impact and high-impact AP ammo in their belts at a ratio of 3:2/4:2. AP ammo has higher armor penetration at the cost of lower overall damage. **Note that AP ammo is also high-impact capable and can still execute**.

AP ammo is available in the CO vendor for 20 points, while standard high-impact is available for 15 points. Non-impact ammo for both the Deagle and the Mateba are found in their civilian/ERT counterparts only.

image
image

The **Mateba** has been rebalanced into a slower but more powerful weapon, having more damage per shot and higher accuracy at the cost of lowered fire rate, higher recoil and lower ammo capacity (6 in the chamber).
```
fire_delay = FIRE_DELAY_TIER_2
burst_amount = BURST_AMOUNT_TIER_3
burst_delay = FIRE_DELAY_TIER_7
accuracy_mult = BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_2
accuracy_mult_unwielded = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_5
scatter = SCATTER_AMOUNT_TIER_7
burst_scatter_mult = SCATTER_AMOUNT_TIER_6
scatter_unwielded = SCATTER_AMOUNT_TIER_2
damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_10
recoil = RECOIL_AMOUNT_TIER_2
recoil_unwielded = RECOIL_AMOUNT_TIER_2
```

The **Desert Eagle** has been rebalanced into a gun that shoots faster, holds more ammo and has lower recoil, but trades that for lower damage per shot, lower accuracy (meaning you have to get close to make all shots land), and shorter stun duration.
```
fire_delay = FIRE_DELAY_TIER_5
accuracy_mult = BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_4
accuracy_mult_unwielded = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_7
scatter = SCATTER_AMOUNT_TIER_6
burst_scatter_mult = SCATTER_AMOUNT_TIER_4
scatter_unwielded = SCATTER_AMOUNT_TIER_3
damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_8
recoil = RECOIL_AMOUNT_TIER_3
recoil_unwielded = RECOIL_AMOUNT_TIER_2
```
Do note that very few of these stats are a _straight_ upgrade from their previous incarnations; numbers have just been shuffled around to make the weapons feel and play differently than each other.

And, yes, this was run through with the CO Council beforehand. I don't see why it wouldn'tve been considering I'm its head.

## Spritesheet
image
image

## Why It's Good For The Game

Currently in the CO's arsenal, the Desert Eagle, by design oversight, is better than the Mateba in most aspects and is the superior option for taking into combat, with the Mateba being more for style points instead of actual use. This PR aims to make both weapons have their own strengths and weaknesses and have them play differently from each other, instead of them feeling samey.

The ammunition rebalance also serves to eliminate the throwaway non-impact Mateba/Deagle ammo that was used, in practice, as simply a last resort for when you ran out of actual high-impact stuff. This replaces that with ammo that's much more useful and pertinent to the weapon at hand, making sure you use the whole belt and not just a fraction of it.

## Testmerge This
Values are arbitrary, and they're likely gonna need quite a bit of tweaking before the weapons' stats both feel right or distinguishable from each other. I don't wanna just make the problem worse, so this should be TM'ed before it actually makes it in. Just in case.

## Changelog

:cl: nauticall
balance: Rebalanced the CO's Mateba and Desert Eagle so that each have their own strengths and weaknesses.
balance: Compared to each other, the Mateba has higher damage per shot and better accuracy at the cost of lower ammo count and lower firerate.
balance: The Deagle, by contrast, fires faster and has more ammo, but has less damage per shot and lower accuracy.
balance: All CO sidearms now come with high-impact ammunition only in their vendors. AP ammo available as a sidegrade.
add: Adds high-impact AP ammo to the Deagle and Mateba that offers much higher armor penetration in exchange for decreased overall damage.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -429,14 +429,22 @@
...
 /datum/ammo/bullet/pistol/heavy/super/highimpact
 	name = ".50 high-impact pistol bullet"
-	penetration = ARMOR_PENETRATION_TIER_2
-	debilitate = list(0,2,0,0,0,1,0,0)
+	penetration = ARMOR_PENETRATION_TIER_1
+	debilitate = list(0,1.5,0,0,0,1,0,0)
 	flags_ammo_behavior = AMMO_BALLISTIC
 
+/datum/ammo/bullet/pistol/heavy/super/highimpact/ap
+	name = ".50 impact armor piercing pistol bullet"
```
Comment (carlarctg) [inline code/modules/projectiles/ammo_datums.dm:437]:
```suggestion
	name = ".50 high-impact armor piercing pistol bullet"
```

consistency with above

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -429,14 +429,22 @@
...
+	penetration = ARMOR_PENETRATION_TIER_10
+	damage = 45
+
 /datum/ammo/bullet/pistol/heavy/super/highimpact/New()
 	..()
 	RegisterSignal(src, COMSIG_AMMO_POINT_BLANK, .proc/handle_battlefield_execution)
 
+/datum/ammo/bullet/pistol/heavy/super/highimpact/ap/New()
```
Comment (carlarctg) [inline code/modules/projectiles/ammo_datums.dm:445]: unnecessary - subtypes already utilize their parents' procs if not overriden. remove this line as New() is called with your intended stats above.

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -690,13 +697,20 @@
 /datum/ammo/bullet/revolver/mateba/highimpact
 	name = ".454 heavy high-impact revolver bullet"
 	debilitate = list(0,2,0,0,0,1,0,0)
-	penetration = ARMOR_PENETRATION_TIER_2
+	penetration = ARMOR_PENETRATION_TIER_1
 	flags_ammo_behavior = AMMO_BALLISTIC
 
+/datum/ammo/bullet/revolver/mateba/highimpact/ap
+	name = ".454 high-impact armor piercing revolver bullet"
```
Comment (carlarctg) [inline code/modules/projectiles/ammo_datums.dm:704]:
```suggestion
	name = ".454 heavy high-impact armor piercing revolver bullet"
```

it stuns just like the parent, should also be heavy for consistency not only with mateba but also m44 heavy IMO

File: code/modules/projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -690,13 +697,20 @@
...
+	penetration = ARMOR_PENETRATION_TIER_10
+	damage = 45
+
 /datum/ammo/bullet/revolver/mateba/highimpact/New()
 	..()
 	RegisterSignal(src, COMSIG_AMMO_POINT_BLANK, .proc/handle_battlefield_execution)
 
+/datum/ammo/bullet/revolver/mateba/highimpact/ap/New()
```
Comment (carlarctg) [inline code/modules/projectiles/ammo_datums.dm:712]: as above , remov

File: code/modules/projectiles/magazines/pistols.dm [hunk 5]
```diff
@@ -157,17 +157,23 @@
...
-	desc = "A heavy variant of Desert Eagle magazines specially-tuned to fit a high-ranking officer. This variant is optimized for anti-armor."
+	desc = "Seven rounds of devastatingly powerful 50-caliber destruction."
 	gun_type = /obj/item/weapon/gun/pistol/heavy/co
 	default_ammo = /datum/ammo/bullet/pistol/heavy/super
 	icon_state = "deagleS"
 
 /obj/item/ammo_magazine/pistol/heavy/super/highimpact
 	name = "\improper High Impact Desert Eagle magazine (.50)"
-	desc = "A heavy variant of Desert Eagle magazines specially-tuned to fit a high-ranking officer. This variant is optimized for anti-personnel. Don't fire it at anyone you'd like to stay alive."
+	desc = "Seven rounds of devastatingly powerful 50-caliber destruction. Packs a devastating punch. This variant is optimized for anti-personnel. Point away from anything you value."
```
Comment (carlarctg) [inline code/modules/projectiles/magazines/pistols.dm:167]: dont use devastatingly twice !!! 


rando sugg:
```suggestion
	desc = "Seven rounds of devastatingly powerful 50-caliber destruction. The bullets are tipped with a synthesized osmium and lead alloy to stagger absolutely anything they hit. Point away from anything you value."
```

File: code/modules/projectiles/magazines/pistols.dm [hunk 6]
```diff
@@ -157,17 +157,23 @@
...
 	default_ammo = /datum/ammo/bullet/pistol/heavy/super/highimpact
 	icon_state = "deagleE"
 
+/obj/item/ammo_magazine/pistol/heavy/super/highimpact/ap
+	name = "\improper High Impact Armor Piercing Desert Eagle magazine (.50)"
+	desc = "Seven rounds of devastatingly powerful 50-caliber destruction. Packs a devastating punch. This variant is optimized against armored targets at the cost of lower overall damage. Point away from anything you value."
```
Comment (carlarctg) [inline code/modules/projectiles/magazines/pistols.dm:173]:
```suggestion
	desc = "Seven rounds of devastatingly powerful 50-caliber destruction. Packs a devastating punch. The bullets are tipped with an osmium-tungsten carbide alloy to not only stagger but shred through any target's armor. Issued in few numbers due to the massive production cost and worries about hull breaches. Point away from anything you value."
```

File: code/modules/projectiles/magazines/revolvers.dm [hunk 7]
```diff
@@ -93,10 +93,16 @@
 
 /obj/item/ammo_magazine/revolver/mateba/highimpact
 	name = "\improper High Impact Mateba speed loader (.454)"
-	desc = "A formidable .454 speedloader, made exclusively for the Mateba autorevolver. Packs a devastating punch. This high impact variant is optimized for anti-personnel. Don't fire this at anyone you want to stay alive."
+	desc = "A formidable .454 speedloader, made exclusively for the Mateba autorevolver. Packs a devastating punch. This high impact variant is optimized for anti-personnel. Don't point at anything you don't want to destroy."
```
Comment (carlarctg) [inline code/modules/projectiles/magazines/revolvers.dm:96]: copy my descs here if u like em for consistency

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 8]
```diff
@@ -13,10 +13,10 @@ GLOBAL_LIST_INIT(cm_vending_gear_commanding_officer, list(
...
 
 		list("SIDEARM AMMUNITION", 0, null, null, null),
-		list("High Impact Mateba Speedloader (.454)", 20, /obj/item/ammo_magazine/revolver/mateba/highimpact, null, VENDOR_ITEM_REGULAR),
-		list("Mateba Speedloader (.454)", 10, /obj/item/ammo_magazine/revolver/mateba, null, VENDOR_ITEM_REGULAR),
-		list("High Impact Desert Eagle Magazine (.50)", 20, /obj/item/ammo_magazine/pistol/heavy/super/highimpact, null, VENDOR_ITEM_REGULAR),
-		list("Heavy Desert Eagle Magazine (.50)", 10, /obj/item/ammo_magazine/pistol/heavy/super, null, VENDOR_ITEM_REGULAR),
+		list("High Impact Mateba Speedloader (.454)", 15, /obj/item/ammo_magazine/revolver/mateba/highimpact, null, VENDOR_ITEM_RECOMMENDED),
+		list("Impact AP Mateba Speedloader (.454)", 20, /obj/item/ammo_magazine/revolver/mateba/highimpact/ap, null, VENDOR_ITEM_REGULAR),
+		list("High Impact Desert Eagle Magazine (.50)", 15, /obj/item/ammo_magazine/pistol/heavy/super/highimpact, null, VENDOR_ITEM_RECOMMENDED),
+		list("Impact AP Desert Eagle Magazine (.50)", 20, /obj/item/ammo_magazine/pistol/heavy/super/highimpact/ap, null, VENDOR_ITEM_REGULAR),
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:19]: shouldn't this be high impact AP

File: code/modules/projectiles/magazines/pistols.dm [hunk 9]
```diff
@@ -157,17 +157,23 @@
...
 
 /obj/item/ammo_magazine/pistol/heavy/super/highimpact
 	name = "\improper High Impact Desert Eagle magazine (.50)"
-	desc = "A heavy variant of Desert Eagle magazines specially-tuned to fit a high-ranking officer. This variant is optimized for anti-personnel. Don't fire it at anyone you'd like to stay alive."
+	desc = "Seven rounds of devastatingly powerful 50-caliber destruction. The bullets are tipped with a synthesized osmium and lead alloy to stagger absolutely anything they hit. Point away from anything you value."
 	default_ammo = /datum/ammo/bullet/pistol/heavy/super/highimpact
 	icon_state = "deagleE"
 
+/obj/item/ammo_magazine/pistol/heavy/super/highimpact/ap
+	name = "\improper High Impact Armor Piercing Desert Eagle magazine (.50)"
```
Comment (stanalbatross) [inline code/modules/projectiles/magazines/pistols.dm:172]: `name = "\improper High Impact Armor-Piercing Mateba speed loader (.454)"`
pick one style for armour-piercing/armor piercing hyphenation

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by ThePiachu — APPROVED** [review]
Code looks good

**Review by fira — APPROVED** [review]

**Review by Agameofscones — APPROVED** [review]
sprites r gud

**Pull request merged**